src/main/cpp/checkpoint/CheckPoint.cpp

```diff
@@ -0,0 +1,675 @@
+	if (exist <= 0) {
+		hid_t temp = H5Gcreate2(m_file, datestr.c_str(), H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+		H5Gclose(temp);
+	}
+
+	hsize_t dims[1] = {pop.size()};
+	hid_t dataspace = H5Screate_simple(1, dims, nullptr);
+	std::string spot = datestr + "/Population";
+
+	hid_t chunkP = H5Pcreate(H5P_DATASET_CREATE);
+	hsize_t chunkDims[1] = {1};
+	H5Pset_chunk(chunkP, 1, chunkDims);
+
+	hid_t newType = H5Tcreate(H5T_COMPOUND, sizeof(personData));
+
+	H5Tinsert(newType, "ID", HOFFSET(personData, ID), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "Age", HOFFSET(personData, Age), H5T_NATIVE_DOUBLE);
+	H5Tinsert(newType, "Gender", HOFFSET(personData, Gender), H5T_NATIVE_CHAR);
+	H5Tinsert(newType, "Participating", HOFFSET(personData, Participating), H5T_NATIVE_HBOOL);
+	H5Tinsert(newType, "Immune", HOFFSET(personData, Immune), H5T_NATIVE_HBOOL);
+	H5Tinsert(newType, "Infected", HOFFSET(personData, Infected), H5T_NATIVE_HBOOL);
+	H5Tinsert(newType, "StartInf", HOFFSET(personData, StartInf), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "EndInf", HOFFSET(personData, EndInf), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "StartSympt", HOFFSET(personData, StartSympt), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "EndSympt", HOFFSET(personData, EndSympt), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "TimeInfected", HOFFSET(personData, TimeInfected), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "Household", HOFFSET(personData, Household), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "School", HOFFSET(personData, School), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "Work", HOFFSET(personData, Work), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "Primary", HOFFSET(personData, Primary), H5T_NATIVE_UINT);
+	H5Tinsert(newType, "Secondary", HOFFSET(personData, Secondary), H5T_NATIVE_UINT);
+
+	hid_t dataset = H5Dcreate2(m_file, spot.c_str(), newType, dataspace, H5P_DEFAULT, chunkP, H5P_DEFAULT);
+
+	chunkP = H5Pcreate(H5P_DATASET_XFER);
+	unsigned int i = 0;
+
+	pop.serial_for([this, &i, &newType, &dataspace, &dataset, &chunkP](const Person& p, unsigned int) {
+		personData data(p);
+
+		hsize_t start[1] = {i};
+		hsize_t count[1] = {1};
+		hid_t chunkspace = H5Screate_simple(1, count, nullptr);
+		H5Sselect_hyperslab(dataspace, H5S_SELECT_SET, start, nullptr, count, nullptr);
+		H5Dwrite(dataset, newType, chunkspace, dataspace, chunkP, &data);
+		i++;
+		H5Sclose(chunkspace);
+	});
+	H5Tclose(newType);
+	H5Dclose(dataset);
+	H5Sclose(dataspace);
+}
+
+void CheckPoint::WriteFileDSet(const std::string& filename, const std::string& setname)
+{
+	boost::filesystem::path path = util::InstallDirs::GetDataDir();
+	boost::filesystem::path filep(filename);
+	boost::filesystem::path fullpath = path / filep;
+	if (!is_regular_file(fullpath)) {
+		FATAL_ERROR("Unable to find file: " + fullpath.string());
+	}
+	std::ifstream f(fullpath.string());
+
+	std::string str((std::istreambuf_iterator<char>(f)), std::istreambuf_iterator<char>());
+	std::vector<char> dataset;
```

Comment:
Why don't you just construct `dataset` from the `std::istreambuf_iterator<char>` values right away instead of first reading the data into an `std::string` and then copying the result into an `std::vector<char>`? The following should work (according to [StackOverflow](http://stackoverflow.com/questions/4423361/constructing-a-vector-with-istream-iterators)) and clarifies your intent IMO:

```cpp
std::vector<char> dataset{std::istreambuf_iterator<char>(f)), std::istreambuf_iterator<char>()};
```